### PR TITLE
Fixes for static pkg-config Windows build

### PR DESCRIPTION
**Pull request by PaulWagener**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Defers the `gdal_i.lib` missing message until after the pkg-config check and outputs pkg-config metadata in case of a static build.

### CODE REVIEW

**Comment by lnicola** [comment]
Bah, this script is so hard to follow.

Can you please add a changelog entry?

**Comment by PaulWagener** [comment]
Done

**Comment by lnicola** [comment]
I can't really verify this, but it seems reasonable.
